File: src/main/java/de/pbauerochse/worklogviewer/fx/components/treetable/SummaryColumn.java
package de.pbauerochse.worklogviewer.fx.components.treetable;

import de.pbauerochse.worklogviewer.util.FormattingUtil;
import javafx.scene.control.TreeTableColumn;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

/**
 * Displays the total amount of time spent
 * for each Issue, as well as a total time
 * spent row
 */
class SummaryColumn extends TreeTableColumn<TreeTableRowModel, TreeTableRowModel> {

    private static final Logger LOGGER = LoggerFactory.getLogger(SummaryColumn.class);

    public SummaryColumn() {
        super(FormattingUtil.getFormatted("view.main.summary"));
        setSortable(false);
//        setCellValueFactory(param -> new SimpleObjectProperty(param.getValue().getValue()));
//        setCellFactory(param -> {
//            TreeTableCell<DisplayRow, DisplayRow> summaryCell = new TreeTableCell<DisplayRow, DisplayRow>() {
//
//                @Override
//                protected void updateItem(DisplayRow item, boolean empty) {
//                    super.updateItem(item, empty);
//
//                    getStyleClass().removeAll(ALL_WORKLOGVIEWER_CLASSES);
//
//                    if (empty) {
//                        setText(StringUtils.EMPTY);
//                    } else {
//                        setText(FormattingUtil.formatMinutes(item.getTotaltimeSpent()));
//                        if (item.isGroupContainer()) {
//                            getStyleClass().add(GROUP_COLUMN_OR_CELL_CSS_CLASS);
//                        } else {
//                            getStyleClass().add(SUMMARY_COLUMN_OR_CELL_CSS_CLASS);
//                        }
//                    }
//                }
//            };
//
//            summaryCell.setAlignment(Pos.CENTER_RIGHT);
//            return summaryCell;
//        });

        setPrefWidth(120);
    }
}

File: src/main/java/de/pbauerochse/worklogviewer/fx/components/treetable/SummaryColumn.kt
package de.pbauerochse.worklogviewer.fx.components.treetable

import de.pbauerochse.worklogviewer.fx.tablecolumns.CellStyleClasses
import de.pbauerochse.worklogviewer.fx.tablecolumns.CellStyleClasses.GROUP_CELL
import de.pbauerochse.worklogviewer.fx.tablecolumns.CellStyleClasses.SUMMARY_CELL
import de.pbauerochse.worklogviewer.util.FormattingUtil
import de.pbauerochse.worklogviewer.util.FormattingUtil.formatMinutes
import javafx.beans.property.SimpleObjectProperty
import javafx.scene.control.TreeTableCell
import javafx.scene.control.TreeTableColumn
import javafx.util.Callback

/**
 * Displays the total amount of time spent
 * for each Issue, as well as a total time
 * spent within the timerange row
 */
internal class SummaryColumn : TreeTableColumn<TreeTableRowModel, TreeTableRowModel>(FormattingUtil.getFormatted("view.main.summary")) {
    init {
        isSortable = false
        cellValueFactory = Callback { col -> SimpleObjectProperty(col.value.value) }
        cellFactory = Callback { _ -> SummaryCell() }
        prefWidth = 120.0
    }
}

class SummaryCell : TreeTableCell<TreeTableRowModel, TreeTableRowModel>() {

    override fun updateItem(item: TreeTableRowModel?, empty: Boolean) {
        super.updateItem(item, empty)

        styleClass.removeAll(CellStyleClasses.ALL_WORKLOGVIEWER_CLASSES)
        tooltip = null
        text = null

        if (!empty && item != null) {
            val timeInMinutes = item.getTotalTimeSpent()
            if (timeInMinutes > 0) {
                text = formatMinutes(timeInMinutes)
            }

            when {
                item.isSummaryRow -> styleClass.add(SUMMARY_CELL)
                item.isIssueRow -> styleClass.add(SUMMARY_CELL)
                item.isGroupByRow -> styleClass.add(GROUP_CELL)
            }
        }
    }
}
